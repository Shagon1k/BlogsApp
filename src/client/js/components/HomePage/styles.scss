.homePage {
	position: absolute;
	max-width: 30%;
	height: 150px;
	box-sizing: border-box;
	padding: 20px;
	margin: auto;
	top: 0;
	bottom: 0;
	right: 0;
	left: 0;
	background: #50626C;
	color: white;

	.homePageInner {
		display: flex;
		flex-flow: column wrap;
		align-items: center;
		justify-content: space-between;
		height: 100%;
		
		.infoText {
			font-size: 18px;
	
			.loginLnk,
			.registerLnk {
				font-size: 18px;
			}
	
			.userName {
				color: #a3bbe0;
			}
		}
	
		.blogsListLnk {
			font-size: 20px;
		}
	}
}