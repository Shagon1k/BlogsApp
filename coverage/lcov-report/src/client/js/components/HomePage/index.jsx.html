<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/HomePage/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/HomePage</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Route, Redirect, Link } from 'react-router-dom';
import './styles.scss';
&nbsp;
const HomePage = ({isLoggedIn, onLogout, user}) =&gt; (
	&lt;div className="homePage"&gt;
		{!isLoggedIn ? 
			&lt;div className="homePageInner"&gt;
				&lt;span className="infoText"&gt;
					Hello! Welcome to BlogsApp!
				&lt;/span&gt;
				&lt;span className="infoText"&gt;
					Please &lt;Link className="lnk loginLnk" to="/login"&gt;Log in&lt;/Link&gt; or &lt;Link className="lnk registerLnk" to="/register"&gt;Register&lt;/Link&gt; to proceed.
				&lt;/span&gt;
			&lt;/div&gt; :
			&lt;div className="homePageInner"&gt;
				&lt;span className="infoText"&gt;
					Hello, &lt;span className="userName"&gt;{user.username}&lt;/span&gt;! Welcome to BlogApps!
				&lt;/span&gt;
				&lt;span&gt;
					Feel free to check &lt;Link className="lnk blogsListLnk" to="/blogs"&gt;Blogs list&lt;/Link&gt;.
				&lt;/span&gt;
				&lt;a href="#" className="lnk logoutLink" onClick={() =&gt; onLogout()}&gt;Log out&lt;/a&gt;
			&lt;/div&gt;
		}
	&lt;/div&gt;
)
&nbsp;
export default HomePage;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="else path not taken" >E</span></span></span></pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 03 2018 15:03:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\sorter.js"></script>
</body>
</html>
