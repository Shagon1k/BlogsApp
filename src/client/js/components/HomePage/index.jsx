import React from 'react';
import { Route, Redirect, Link } from 'react-router-dom';
import './styles.scss';

const HomePage = ({isLoggedIn, onLogout, user}) => (
	<div className="homePage">
		{!isLoggedIn ? 
			<div className="homePageInner">
				<span className="infoText">
					Hello! Welcome to BlogsApp!
				</span>
				<span className="infoText">
					Please <Link className="lnk loginLnk" to="/login">Log in</Link> or <Link className="lnk registerLnk" to="/register">Register</Link> to proceed.
				</span>
			</div> :
			<div className="homePageInner">
				<span className="infoText">
					Hello, <span className="userName">{user.username}</span>! Welcome to BlogApps!
				</span>
				<span>
					Feel free to check <Link className="lnk blogsListLnk" to="/blogs">Blogs list</Link>.
				</span>
				<a href="#" className="lnk logoutLink" onClick={() => onLogout()}>Log out</a>
			</div>
		}
	</div>
)

export default HomePage;