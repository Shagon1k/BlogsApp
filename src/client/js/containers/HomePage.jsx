import { connect } from 'react-redux';
import HomePage from '../components/HomePage/index.jsx';
import { requestLogout } from '../actions'

const mapStateToProps = state => ({
	isLoggedIn: state.authentication.isLoggedIn,
	user: state.authentication.user
})

const mapDispatchToProps = dispatch => ({
	onLogout: () => {
		dispatch(requestLogout());
	}
})

const HomePageContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(HomePage)

export default HomePageContainer;