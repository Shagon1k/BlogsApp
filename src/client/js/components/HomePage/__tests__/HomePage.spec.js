import React from 'react';
import HomePage from '../index.jsx';

describe('HomePage component', () => {
	const user = {
			username: 'Vasya',
			email: 'vasya@gmail.com'
		};
	const onLogout = jest.fn();

	it('renders "Please Log In or Register message" if user is not logged in', () => {
		const wrapper = shallow(<HomePage isLoggedIn={false} onLogout={onLogout} user={null}/>)
		expect(wrapper).toMatchSnapshot();

		wrapper.find('.logoutLink').simulate('click');
		expect(onLogout).toBeenCalled();
	});
	
	it('renders "Welcome to BlogsApp" message if user is logged in', () => {
		const wrapper = shallow(<HomePage isLoggedIn={true} onLogout={onLogout} user={user}/>);
		expect(wrapper).toMatchSnapshot();
	})
})