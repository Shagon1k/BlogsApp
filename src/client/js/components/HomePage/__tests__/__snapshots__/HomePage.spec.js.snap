// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders "Please Log In or Register message" if user is not logged in 1`] = `
<StaticRouter
  basename=""
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HomePage
      isLoggedIn={false}
      onLogout={[MockFunction]}
      user={null}
    >
      <div
        className="homePage"
      >
        <div
          className="homePageInner"
        >
          <span
            className="infoText"
          >
            Hello! Welcome to BlogsApp!
          </span>
          <span
            className="infoText"
          >
            Please 
            <Link
              className="lnk loginLnk"
              replace={false}
              to="/login"
            >
              <a
                className="lnk loginLnk"
                href="/login"
                onClick={[Function]}
              >
                Log in
              </a>
            </Link>
             or 
            <Link
              className="lnk registerLnk"
              replace={false}
              to="/register"
            >
              <a
                className="lnk registerLnk"
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
            </Link>
             to proceed.
          </span>
        </div>
      </div>
    </HomePage>
  </Router>
</StaticRouter>
`;

exports[`HomePage component renders "Welcome to BlogsApp" message if user is logged in 1`] = `
<div
  className="homePage"
>
  <div
    className="homePageInner"
  >
    <span
      className="infoText"
    >
      Hello, 
      <span
        className="userName"
      >
        Vasya
      </span>
      ! Welcome to BlogApps!
    </span>
    <span>
      Feel free to check 
      <Link
        className="lnk blogsListLnk"
        replace={false}
        to="/blogs"
      >
        Blogs list
      </Link>
      .
    </span>
    <a
      className="lnk logoutLink"
      href="#"
      onClick={[Function]}
    >
      Log out
    </a>
  </div>
</div>
`;
